docs/source/metrics.rst

```diff
@@ -393,11 +443,15 @@ class.
 First, write your own metric function, similar to the existing ones with required
 arguments ``forecast``, ``observations``, ``dim=None``, and ``**metric_kwargs``::
 
-  from climpred.metrics import Metric
+  from climpred.metrics import Metric, _sanitize_kwargs, _rename_dim
 
   def _my_msle(forecast, observations, dim=None, **metric_kwargs):
       """Mean squared logarithmic error (MSLE).
       https://peltarion.com/knowledge-center/documentation/modeling-view/build-an-ai-model/loss-functions/mean-squared-logarithmic-error."""
+      # process dim and metric_kwargs; many not be required
```

Comment:
What does "many not be required" mean? I think for a user it should be clear if they need to include this. So might just say "required pre-processing for metrics. make sure to include these lines." I wonder if there's any way to avoid having the user use private (`_`) funcs like this in the custom metrics. This is fine for now though.

Reply:
this will be handled in the cleanup PR